### PR TITLE
[MRG] BUG: fix offset in morlet wavelets

### PR DESCRIPTION
**Pull request by rythorpe**

### CODE REVIEW

File: spectralevents.py [hunk 1]
```diff
@@ -195,13 +195,15 @@ def _energyvec(f, s, Fs, width=7.):
     sf = f / width
     st = 1 / (2 * np.pi * sf)
 
-    t = np.arange(-3.5 * st, 3.5 * st, dt)
-    m = _morlet(f, t, width)
+    t_single_side = np.arange(0., 3.5 * st, dt)
+    # mirror about 0; always an odd # of elements
+    times = np.r_[-t_single_side[-1:0:-1], t_single_side]
```
Comment (rythorpe) [inline spectralevents.py:200]: Note that previously, the center of the Morlet wavelet (i.e., at time 0) wasn't necessarily aligned with the signal sample point during convolution.

File: spectralevents.py [hunk 2]
```diff
@@ -195,16 +195,17 @@ def _energyvec(f, s, Fs, width=7.):
...
+    wavelet = _morlet(f, times, width)
 
-    y = np.convolve(s, m)
-    y = 2 * (dt * np.abs(y)) ** 2
-    lowerLimit = int(np.ceil(len(m) / 2))
-    upperLimit = int(len(y) - np.floor(len(m) / 2) + 1)
```
Comment (rythorpe) [inline spectralevents.py:204]: To be consistent with the original matlab version, `lowerLimit` would have needed to be `int(np.ceil(len(m) / 2) - 1)` given Pythonic indexing. Similar for `upperLimit`. The matlab version wasn't totally correct though either....

**Comment by rythorpe** [comment]
@darcywaller would you mind double checking my logic here?

**Comment by darcywaller** [comment]
From a precursory look at changes, that makes sense to me! I can take a further look and replicate line-by-line a little later.

File: spectralevents_ts2tfr.m [hunk 3]
```diff
@@ -67,12 +67,13 @@
...
 
-t=-3.5*st:dt:3.5*st;
+t=0:dt:3.5*st;
+t=[-t(end:-1:2),t];
 m = morlet(f,t,width);
 
 y = conv(s,m);
 y = 2*(dt*abs(y)).^2;
-y = y(ceil(length(m)/2):length(y)-floor(length(m)/2));
+y = y(floor(length(m)/2) + 1:length(y)-floor(length(m)/2));
```
Comment (darcywaller) [inline spectralevents_ts2tfr.m:76]: Checking - is this line for readability purposes/matching python code? I'm pretty sure I'm getting the same result whether I use ceil(length(m)/2) or floor(length(m)/2) + 1.

File: spectralevents.py (hunk 2 above)
Comment (darcywaller) [reply spectralevents.py:204]: Yup - that would do it! This procedure for rounding down half the (odd) length looks good.

File: spectralevents.py (hunk 1 above)
Comment (darcywaller) [reply spectralevents.py:200]: Got it, this edit makes sense then for forcing wavelet alignment to timepoint 0 (alongside the parallel change in the Matlab script).

**Comment by darcywaller** [comment]
OK - makes perfect sense me after a little bit more interaction with the code! If I'm understanding correctly, those two issues together (wavelet unaligned with timepoint 0, issues with rounding) give us our two sample points exactly by which the two versions were offset, right?

File: spectralevents_ts2tfr.m (hunk 3 above)
Comment (rythorpe) [reply spectralevents_ts2tfr.m:76]: Those two expressions are only equal when `length(m)` is odd. For example, when `length(m)` = `2`, `ceil(2/2)` = `1` whereas `floor(2/2) + 1` = `1`.

**Comment by rythorpe** [comment]
> OK - makes perfect sense me after a little bit more interaction with the code! If I'm understanding correctly, those two issues together (wavelet unaligned with timepoint 0, issues with rounding) give us our two sample points exactly by which the two versions were offset, right?

Yes, I believe so! I'm going to go ahead and rebase and merge so that we can move forward with your PR.

**Pull request merged**